docs/reference/functions.md

```diff
@@ -101,6 +101,32 @@ notMatch | if provided, value must _not_ match this regex | no
         notMatch: ".+\/$"
 ```
 
+## casing
+
+Regular expressions! 
+
+<!-- title: functionOptions -->
+
+name | description | required?
+---------|----------|---------
+type | the casing type to match against | yes
+disallowDigits | if not truthy, digits are allowed | no
+
+<!-- title: example -->
+
+```yaml
+camel-case-name:
+  description: Name should camelCased.
+  type: style
+  given: "$.name"
+  then:
+    function: casing
+    functionOptions:
+      type: camel
+```
+
+Available types are: flat, camel, pascal, kebab, cobol, snake, macro.
```

Comment:
Oops here they are. Could you move this up near the description and provide examples of each?

Reply:
Do you mean to specify valid and invalid examples for each casing type?